api/dao/reaperutil.py

```diff
@@ -144,4 +157,51 @@ def create_container_hierarchy(metadata):
         upsert=True,
         return_document=pymongo.collection.ReturnDocument.AFTER,
     )
-    return ReapedAcquisition(acquisition_obj, file_)
+    return TargetAcquisition(acquisition_obj, file_)
+
+def update_container_hierarchy(metadata, acquisition_id, level):
+    project = metadata.get('project')
+    session = metadata.get('session')
+    acquisition = metadata.get('acquisition')
+    now = datetime.datetime.utcnow()
+    if acquisition.get('timestamp'):
+        acquisition['timestamp'] = dateutil.parser.parse(acquisition['timestamp'])
+    acquisition['modified'] = now
+    acquisition_obj = _update_container({'_id': acquisition_id}, acquisition, 'acquisitions')
+    if acquisition_obj is None:
+        raise APIStorageException('acquisition doesn''t exist')
+    if acquisition.get('timestamp'):
+        session_obj = config.db.sessions.find_one_and_update(
+            {'_id': acquisition_obj['session']},
+            {
+                '$min': dict(timestamp=acquisition['timestamp']),
+                '$set': dict(timezone=acquisition.get('timezone'))
+            },
+            return_document=pymongo.collection.ReturnDocument.AFTER
+        )
+        config.db.projects.find_one_and_update(
+            {'_id': session_obj['project']},
+            {
+                '$max': dict(timestamp=acquisition['timestamp']),
+                '$set': dict(timezone=acquisition.get('timezone'))
+            }
+        )
+    session_obj = None
+    if session:
+        session['modified'] = now
+        session_obj = _update_container({'_id': acquisition_obj['session']}, session, 'sessions')
+    if project:
+        project['modified'] = now
+        if not session_obj:
+            session_obj = config.db.sessions.find_one({'_id': acquisition_obj['session']})
+        _update_container({'_id': session_obj['project']}, project, 'projects')
```

Comment:
It's simply bad.
We would need a general "transaction" for multiple mongo operations as the reaper endpoint also applies multiple updates serially.

WIredTiger,  a new mongo driver (default in 3.2) seems to provide a transaction system, http://source.wiredtiger.com/2.4.1/transactions.html

However I think that this issue is not in the scope of this PR, but we should definitely address it sooner than later. We should explore if WiredTiger is the way to go or if a home baked system would be better in term of flexibility/optimization vs development/maintenance time.